### PR TITLE
Active

### PR DESCRIPTION
**Pull request by unkrich**
Change updates the jobs to take a start_date so that we do not do repeat work. 

For now the default lookback days is hardcoded and small because the job takes very long to execute. This will be updated later.

### CODE REVIEW

**Comment by unkrich** [comment]
This update also includes a pipeline for sending alerts to integrations destinations

File: src/scheduler/base.py [hunk 1]
```diff
@@ -72,12 +78,17 @@ def jobstore(self):
 
     @classmethod
     def run_job(cls, job_class_path, job_id, db_url, *args, **kwargs):
+        datasource_id = args[0]
+
         jobstore = MsiJobStore(url=db_url)
+        last_run = jobstore.get(datasource_id).get('created_at')
+        print(last_run)
```
Comment (iporollo) [inline src/scheduler/base.py:85]: remove

File: src/server/integrations/slack.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                    "type": "string",
+                    "title": "Slack Webhook URL",
+                },
+            }
+        }
+
+    @classmethod
+    def _create_request_data(cls, metric):
+        body = {
+            "text": "Danny Torrence left a 1 star review for your property.",
```
Comment (iporollo) [inline src/server/integrations/slack.py:23]: what?

File: src/server/jobs/metadata.py [hunk 3]
```diff
@@ -29,10 +29,17 @@ def meta_info(cls):
...
+        integrations = db.session.query(Integration).all()
         source = db.session.query(DataSource).filter(DataSource.id == datasource_id).one()
         source_configuration = source.to_dict()
         source_configuration['config']['type'] = source_configuration['type']
 
+        source_configuration['config']['start_date'] = str(self.last_run)
+        destination['start_date'] = str(self.last_run)
+
+        print(source_configuration['config'])
+        print(destination)
```
Comment (iporollo) [inline src/server/jobs/metadata.py:41]: remove?

File: src/server/output.txt [hunk 4]
```diff
@@ -0,0 +1,442 @@
+ * Environment: production
```
Comment (iporollo) [inline src/server/output.txt:1]: we need to gitignore this file. Also need to change the password.

**Review by iporollo — CHANGES_REQUESTED** [review]

File: src/ingestion/sources/base.py [hunk 5]
```diff
@@ -33,6 +35,30 @@ def database(self):
     def schema(self):
         return json.loads(self.configuration).get('schema')
 
+    def _start_date(self):
+        return json.loads(self.configuration)['start_date']
+
+    def minutes_ago(self):
+        thirty_days_ago = -int(.25*24*60)
```
[the remark this answers is not in the document]
Comment (unkrich) [reply src/ingestion/sources/base.py:42]: read initial description

File: src/scheduler/base.py (hunk 1 above)
Comment (unkrich) [reply src/scheduler/base.py:85]: still important as I'm not 100% sure this works.

File: src/server/integrations/slack.py (hunk 2 above)
Comment (unkrich) [reply src/server/integrations/slack.py:23]: He wasn't very happy

File: src/server/jobs/metadata.py (hunk 3 above)
Comment (unkrich) [reply src/server/jobs/metadata.py:41]: yeah you right

File: src/server/output.txt (hunk 4 above)
Comment (unkrich) [reply src/server/output.txt:1]: yep this shouldn't be here. deleting and resetting pass

File: src/scheduler/base.py (hunk 1 above)
Comment (unkrich) [reply src/scheduler/base.py:85]: Removed

File: src/ingestion/sources/base.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (unkrich) [reply src/ingestion/sources/base.py:42]: Keeping this for now.

File: src/ui/src/pages/app/monitors/Detail/components/ExecutionsTable.tsx [hunk 6]
```diff
@@ -7,29 +7,29 @@ import paginationFactory from "react-bootstrap-table2-paginator";
 
 import ToolkitProvider from 'react-bootstrap-table2-toolkit';
 
-const ExecutionsTable: React.FC<{monitor_id: any}> = ({ monitor_id }) => {
+const ExecutionsTable: React.FC<{datasource_id: any}> = ({ datasource_id }) => {
```
Comment (iporollo) [inline src/ui/src/pages/app/monitors/Detail/components/ExecutionsTable.tsx:10]: ultimately prefer this to be camel case but not blocking

File: src/ingestion/sources/base.py [hunk 7]
```diff
@@ -33,6 +35,30 @@ def database(self):
...
+
+        start_date = self._start_date()
+        if start_date is None:
+            print('no start date')
```
Comment (iporollo) [inline src/ingestion/sources/base.py:46]: should be a log but not blocking

File: src/ingestion/sources/base.py [hunk 8]
```diff
@@ -33,6 +35,30 @@ def database(self):
...
+            return thirty_days_ago
+
+        try:
+            try:
+                start_date = parser.parse(start_date)
+            except:
+                logging.info("Start date is not a string: {}", str(start_date))
+
+            print("Starting from {}", -int((datetime.now() - start_date).total_seconds() / 60))
```
Comment (iporollo) [inline src/ingestion/sources/base.py:55]: since we have it logging, no need for a print statement

**Review by iporollo — APPROVED** [review]

**Pull request merged**